imxrt-hal/src/uart.rs

```diff
@@ -76,61 +76,87 @@ impl Unclocked {
 
         // -----------------------------------------
         // Disable clocks before modifying selection
-        ccm.ccgr5.modify(|_, w| unsafe {
-            w.cg12()
-                .bits(0b00) // UART1
-                .cg13()
-                .bits(0b00) // UART7
-        });
-        ccm.ccgr0.modify(|_, w| unsafe {
-            w.cg14()
-                .bits(0b00) // UART2
-                .cg6()
-                .bits(0b00) // UART3
-        });
-        ccm.ccgr1.modify(|_, w| unsafe { w.cg12().bits(0b00) }); // UART4
-        ccm.ccgr3.modify(|_, w| unsafe {
-            w.cg1()
-                .bits(0b00) // UART5
-                .cg3()
-                .bits(0b00) // UART6
-        });
-        ccm.ccgr6.modify(|_, w| unsafe { w.cg7().bits(0b00) }); // UART8
-
+        ral::modify_reg!(
+            ral::ccm,
+            ccm,
+            CCGR5,
+            CG12: 0,    // UART1
+            CG13: 0     // UART7
+        );
+        ral::modify_reg!(
+            ral::ccm,
+            ccm,
+            CCGR0,
+            CG14: 0,    // UART2
+            CG6: 0      // UART3
+        );
+        ral::modify_reg!(
+            ral::ccm,
+            ccm,
+            CCGR1,
+            CG12: 0     // UART4
+        );
+        ral::modify_reg!(
+            ral::ccm,
+            ccm,
+            CCGR3,
+            CG1: 0,     // UART5
+            CG3: 0      // UART6
+        );
+        ral::modify_reg!(
+            ral::ccm,
+            ccm,
+            CCGR6,
+            CG7: 0      // UART8
+        );
         // -----------------------------------------
 
         // -------------------------
         // Select clocks & prescalar
-        ccm.cscdr1.write(|w| {
-            w.uart_clk_sel()
-                .variant(clock_select.into())
-                .uart_clk_podf()
-                .variant(prescalar)
-        });
+        ral::modify_reg!(
+            ral::ccm,
+            ccm,
+            CSCDR1,
+            UART_CLK_SEL: (clock_select as u32),
+            UART_CLK_PODF: (prescalar as u32)
+        );
```

Comment:
Noting the change from a `write` to a `modify`. Previous implementation was incorrect, and it would have squashed the values in other fields of `CSCDR1`.